tests/env/components/dns_resolver.go

```diff
@@ -33,28 +33,38 @@ func (h *handler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
 	msg := dns.Msg{}
 	msg.SetReply(r)
 	glog.Infof("dns query:\n%+v", r)
-	switch r.Question[0].Qtype {
-	case dns.TypeA:
-		msg.Authoritative = true
-		domain := msg.Question[0].Name
-
-		address, ok := h.records[domain]
-		if ok {
-			msg.Answer = append(msg.Answer, &dns.A{
-				Hdr: dns.RR_Header{Name: domain, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
-				A:   net.ParseIP(address),
-			})
-			glog.Infof("dns answer: %+v", msg.Answer)
-		} else {
-			msg.Rcode = dns.RcodeNameError
-			glog.Infof("dns return code: %v", dns.RcodeToString[msg.Rcode])
+	msg.Authoritative = true
+	domain := msg.Question[0].Name
+	addresses, ok := h.records[domain]
+	if ok {
+		for _, address := range addresses {
+			ip := net.ParseIP(address)
+			if ip.To4() == nil {
+				// address is IPv6
+				if r.Question[0].Qtype != dns.TypeA {
+					msg.Answer = append(msg.Answer, &dns.AAAA{
+						Hdr:  dns.RR_Header{Name: domain, Rrtype: dns.TypeAAAA, Class: dns.ClassINET, Ttl: 60},
+						AAAA: ip,
+					})
+				}
+			} else {
+				if r.Question[0].Qtype != dns.TypeAAAA {
```

Comment:
If this is false, should we return DNS error instead of empty response?

Reply:
The error should be thrown elsewhere.